agcaballero/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,64 +14,82 @@
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.joda.time.DateTimeZone;
+import org.joda.time.LocalDateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
 
 /**
  * Servlet that accepts data from the comments form &
  * posts that data to the comments.html page
  */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private class Comment {
-    private final String name; // stores name of commenter
-    private final String email; // stores email of commenter
-    private final String comment; // comment itself
-
-    // constructs a Comment object
-    public Comment(String name, String email, String comment) {
-      this.name = name;
-      this.email = email;
-      this.comment = comment;
-    }
-
-    // returns comment in format: "name (email): comment"
-    public String toString() {
-      return name + " (" + email + "): " + comment;
-    }
-  }
-
-  // holds all comments that have been submitted in session so far
-  private final ArrayList<Comment> comments = new ArrayList<Comment>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Retrieve from Datastore entities of type "Comment"
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
     StringBuilder commentDivs = new StringBuilder();
 
-    // build a String of divs to add to page
-    for (Comment comment : comments) {
-      commentDivs.append("<div class='comment-div'><p>" + comment + "</p></div>");
+    // Build a String of divs to hold comments to add to page
+    for (Entity entity : results.asIterable()) {
+      // Retrieve info from the Entity
+      Date timestamp = (Date) entity.getProperty("timestamp");
+      String name = (String) entity.getProperty("name");
+      String email = (String) entity.getProperty("email");
+      String comment = (String) entity.getProperty("comment");
+
+      // append current div to HTML string commentDivs
+      commentDivs.append(formatComment(timestamp, name, email, comment));
     }
 
-    // respond with the commentDivs html
+    // Respond to request with the commentDivs html
     response.setContentType("application/html;");
     response.getWriter().println(commentDivs);
   }
 
+  // Takes properties of a comment and formats them with proper HTML
+  private String formatComment(Date timestamp, String name, String email, String comment) {
+    // gets time in local time zone (default: US ET)
+    LocalDateTime ldt = new LocalDateTime(timestamp.getTime(), DateTimeZone.forID("US/Eastern"));
+    DateTimeFormatter fmt = DateTimeFormat.forPattern("h:mm a M/dd/yy");
```

Comment:
In general non-standard abbreviations are discouraged, as they slow down reading. I'd suggest spelling these out instead (e.g. `dateTime` and `formatter`).

Reply:
Done